src/types/utils.ts

```diff
@@ -268,14 +310,36 @@ export function typeNameToSpecializedTypeNode(astT: TypeName, loc: DataLocation)
     return specializeType(typeNameToTypeNode(astT), loc);
 }
 
+export function inferVariableDeclLocation(decl: VariableDeclaration): DataLocation {
+    if (decl.stateVariable) {
+        return DataLocation.Storage;
+    } else if (decl.storageLocation !== DataLocation.Default) {
+        return decl.storageLocation;
+    } else if (
+        decl.parent instanceof ParameterList ||
+        decl.parent instanceof VariableDeclarationStatement
+    ) {
+        // In 0.4.x param/return locations may be omitted. We assume memory by default.
+        return DataLocation.Memory;
+    } else if (decl.parent instanceof StructDefinition) {
+        return DataLocation.Default;
+    } else if (decl.parent instanceof SourceUnit) {
+        // Global vars don't have a location (no ref types yet)
+        return DataLocation.Default;
+    } else {
+        throw new Error(`NYI variable declaration ${pp(decl)}`);
+    }
```

Comment:
Better to unnest this

Reply:
Fixed in [51bdea0](https://github.com/ConsenSys/solc-typed-ast/pull/142/commits/51bdea0cbff6131127af6ceb6d1e12c11faaecc5).